azkaban-hadoop-security-plugin/src/main/java/azkaban/security/commons/HadoopSecurityManager.java

```diff
@@ -101,6 +102,9 @@ public abstract FileSystem getFSAsUser(String realIdentity, String proxyUser)
   public abstract void cancelTokens(File tokenFile, String userToProxy,
       Logger logger) throws HadoopSecurityManagerException;
 
+  public abstract Credentials getTokens(File tokenFile, Logger logger)
+      throws HadoopSecurityManagerException;
+
```

Comment:
HSM expose a new method to get tokens from file, and implemented in AbstractHSM, which will be inherent by KsudoHSM.